views/index.py

```diff
@@ -325,6 +327,59 @@ def reserve(copy_id):
     return redirect(url_for('library.index'))
 
 
+@library.route('/remove_item/<int:item_id>', methods=['GET', 'POST'])
+def remove_item(item_id):
+    try:
+        user = User.query.get(session['id'])
+        admin = user.has_role('ADMIN')
+    except KeyError:
+        abort(401)
+    except Exception:
+        abort(500)
+    form = RemoveForm()
+    item = LibraryItem.query.get_or_404(item_id)
+    if form.validate_on_submit():
+        db.session.delete(item)
+        db.session.commit()
+        flash('Item has been removed', 'success')
+    authors_list = []
+    if item.type == 'book':
+        authors_list = item.authors_string
+    return render_template('remove_item.html',
+                           form=form,
+                           item=item,
+                           authors_list=authors_list,
+                           admin=admin)
+
+
+@library.route('/remove_copy/<int:item_id>/<string:asset_code>',
+               methods=['GET', 'POST'])
+def remove_copy(item_id, asset_code):
+    try:
+        user = User.query.get(session['id'])
+        admin = user.has_role('ADMIN')
+    except KeyError:
+        abort(401)
+    except Exception:
+        abort(500)
+    form = RemoveForm()
+    item = LibraryItem.query.get_or_404(item_id)
+    copy = Copy.query.filter_by(asset_code=asset_code).first_or_404()
```

Comment:
Not better just pass to the view `copy.id`? As there could be copies without assets, plus `query.get(id)` is more efficient approach than `query.filter_by(...)`

Reply:
Yep, I've tried to handle it this way, but had some obstacles. 
While the asset code does not exist (which tends to happen while we have only one copy of a book or magazine), the whole item is removed, so in the end I think it works properly.